### PR TITLE
Bwa-mem2 linux aarch64 and re-trying osx-64

### PR DESCRIPTION
**Pull request by dslarm**
This extends https://github.com/bioconda/bioconda-recipes/pull/51942 - the OS/X x86 failed to build, and it's not clear when that last worked or whether an underlying OS change is responsible.

This PR does:
- nothing code-wise for osx-64 - it bumps the locked version of compiler from 14 to 15 - which is shown to work locally on my Macbook (Apple Silicon) running Sonoma 14.6.1 with a conda osx-64 (x86) environment.
- use the prior commits for adding the patch to allow compilation on linux-aarch64

NB: at least in Sonoma / locally: version 14 of Apple clang misses typedefs; version 16 errors on use of 'register' - as it is now c++17 by default where 'register' is not allowed.

### CODE REVIEW

**Comment by dslarm** [comment]
@BiocondaBot please add label

**Comment by dslarm** [comment]
@martin-g - can you review?

File: recipes/bwa-mem2/build.sh [hunk 1]
```diff
@@ -9,7 +9,24 @@ if [[ $OSTYPE == "darwin"* ]]; then
     sed -i.bak 's/memset_s/memset8_s/g' ext/safestringlib/safeclib/memset_s.c
     sed -i.bak 's/memset_s/memset8_s/g' ext/safestringlib/safeclib/wmemset_s.c
 fi
-LIBS="${LDFLAGS}" make CC="${CC}" CXX="${CXX}" multi
+
+case "$(uname -m)" in
+  x86_64)
+      LIBS="${LDFLAGS}" make CC="${CC}" CXX="${CXX}" multi ;;
```
Comment (martin-g) [inline recipes/bwa-mem2/build.sh:15]:
```suggestion
      LIBS="${LDFLAGS}" make -j${CPU_COUNT} CC="${CC}" CXX="${CXX}" multi ;;
```

File: recipes/bwa-mem2/build.sh [hunk 2]
```diff
@@ -9,7 +9,24 @@ if [[ $OSTYPE == "darwin"* ]]; then
...
+  aarch64)
+      if [ "$(uname -s)" == Darwin ]
+      then
+        LIBS="${LDFLAGS}" make CC="${CC}" CXX="${CXX}" multi
```
Comment (martin-g) [inline recipes/bwa-mem2/build.sh:19]:
```suggestion
        LIBS="${LDFLAGS}" make -j${CPU_COUNT} CC="${CC}" CXX="${CXX}" multi
```

File: recipes/bwa-mem2/build.sh [hunk 3]
```diff
@@ -9,7 +9,24 @@ if [[ $OSTYPE == "darwin"* ]]; then
...
+      else
+        mkdir ext/simde
+        wget https://github.com/simd-everywhere/simde-no-tests/archive/refs/tags/v0.8.2.tar.gz -O - | tar -xvz
+        mv simde-no-tests-0.8.2/* ext/simde/
+        LIBS="${LDFLAGS}" make arch=native EXE=bwa-mem2 CC="${CC}" CXX="${CXX}" all
```
Comment (martin-g) [inline recipes/bwa-mem2/build.sh:24]:
```suggestion
        LIBS="${LDFLAGS}" make -j${CPU_COUNT} arch=native EXE=bwa-mem2 CC="${CC}" CXX="${CXX}" all
```

File: recipes/bwa-mem2/build.sh [hunk 4]
```diff
@@ -9,7 +9,24 @@ if [[ $OSTYPE == "darwin"* ]]; then
...
+      fi
+      ;;
+  *)
+      echo "Error detecting architecture" ;;
```
Comment (martin-g) [inline recipes/bwa-mem2/build.sh:28]:
```suggestion
      echo "Not supported architecture: $(uname -m)" ;;
```

File: recipes/bwa-mem2/simde.patch [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
++    }
++  }
++  struct timeval tv;
++  gettimeofday(&tv, nullptr);
++  return static_cast<int64_t>(tv.tv_sec) * 1000000 + tv.tv_usec;
++}
++#else
++#error __ARM_ARCH < 6 does not have a standard cyclecount
++#endif
++#endif
```
Comment (martin-g) [inline recipes/bwa-mem2/simde.patch:247]: ❤️

Comment (dslarm) [reply recipes/bwa-mem2/simde.patch:247]: Keeping the patch aligned with the PR that's over in bwa-mem2 github so trying not to change too much.  But if someone does try to build this on a microcontroller.. good luck!  ;-)

**Review by martin-g — APPROVED** [review]

**Pull request merged**